backend/src/main/java/de/neuefische/capstone/plantcomponents/PlantService.java

```diff
@@ -3,21 +3,46 @@
 import lombok.RequiredArgsConstructor;
 import org.springframework.stereotype.Service;
 
-import java.security.Principal;
 import java.util.List;
+import java.util.Optional;
 
 @RequiredArgsConstructor
 @Service
 public class PlantService {
 
     private final PlantRepo plantRepo;
 
-    public Plant createPlant (Plant plant, Principal principal){
-        plant.setUser(principal.getName());
+    public Plant createPlant (Plant plant, String username){
+        plant.setUser(username);
         return plantRepo.save(plant);
     }
 
-    public List<Plant> getAllPlantsByUser(Principal principal) {
-        return plantRepo.findAllByUser(principal.getName());
+    public List<Plant> getAllPlantsByUser(String username) {
+        return plantRepo.findAllByUser(username);
     }
+
+    public List<Plant> getMatchingPlant(String searchedPlant, String username) {
+        return plantRepo.findAllByUser(username).stream()
+                .filter(plant -> plant.getNonScName().toLowerCase().contains(searchedPlant.toLowerCase()))
+                .toList();
+    }
+
+    public Plant deletePlant(String id, String username) {
+        Optional<Plant> optionalPlant = plantRepo.findByIdAndUser(id, username);
+        Plant plant = optionalPlant.orElseThrow(() -> new IllegalArgumentException("No plant with ID: " + id + " found!"));
```

Comment:
You could return an empty `Optional` if no plant was found and use `ResponseEntity.of` in the controller to return a 404 if no plant was found.